hms-api/__tests__/service/idea_comment-service.test.ts

```diff
@@ -0,0 +1,61 @@
+import * as userRepository from '../../src/repository/user-repository';
+import * as ideaCommentRepository from '../../src/repository/idea_comment-repository';
+import * as ideaRepository from '../../src/repository/idea-repository';
+import {
+  createComment,
+  getCommentResponse,
+} from '../../src/service/idea_comment-service';
+import {uuid} from '../../src/util/Uuid';
+import ReferenceNotFoundError from '../../src/error/ReferenceNotFoundError';
+import {randomIdea} from '../repository/domain/idea-maker';
+import {
+  IdeaCommentData,
+  makeIdeaComment,
+} from '../repository/domain/ideaComment-maker';
+import CommentResponse from '../../src/rest/Comment/CommentResponse';
+import {randomUser} from '../repository/domain/user-maker';
+
+const mockPutComment = jest.fn();
+jest
+  .spyOn(ideaCommentRepository, 'putComment')
+  .mockImplementation(mockPutComment);
+const mockUserExists = jest.fn();
+jest.spyOn(userRepository, 'userExists').mockImplementation(mockUserExists);
+const mockIdeaExists = jest.fn();
+jest.spyOn(ideaRepository, 'ideaExists').mockImplementation(mockIdeaExists);
+const mockGetUser = jest.fn();
+jest.spyOn(userRepository, 'getUser').mockImplementation(mockGetUser);
+const mockGetIdea = jest.fn();
+jest.spyOn(ideaRepository, 'getIdea').mockImplementation(mockGetIdea);
+const mockGetComment = jest.fn();
+jest
+  .spyOn(ideaCommentRepository, 'getComment')
+  .mockImplementation(mockGetComment);
+
+describe('Create Idea Comment', () => {
+  test('Missing User', async () => {
```

Comment:
should also write a happy path test as well as the other failure states

Reply:
👁️ 👁️

Reply:
weird authentification token problem when writing happy path and others...maybe you can give me a hint for that in discord

Reply:
I doubt it is anything to do with the auth token as we never actually use the rest layer in these unit tests here.  I do see an issue with the `edit.test.ts` file where you call `new ideaCommentEditRequest()`.  You pass in both the id and the text to it, but it should only get the id

Reply:
will add missing test tommorrow